locust_tasks/locustfile.py

```diff
@@ -166,49 +189,163 @@ def correct_address(self):
 
-    @task(6)
+    @task(7)
     def confirm_mobile_number(self):
-        self.client.post("/request-access-code/confirm-mobile", {
+        """
+        POST 'yes' to confirm mobile number
+        """
+        with self.client.post("/en/requests/access-code/confirm-mobile/", {
             'request-mobile-confirmation': 'yes'
-        })
+        }, catch_response=True) as response:
+            verify_response('RequestUacSms-ConfirmMobileNumber', self, response, 200, Page.CODE_SENT)
+
+
+class RequestNewCodePost(SequentialTaskSet):
+
+    # All users arrive at the start page
+    @task(1)
+    def start_page(self):
+        """
+        GET Start page
+        """
+        self.case = get_next_case()
+        with self.client.get('/en/start/', catch_response=True) as response:
+            verify_response('RequestUacPost-Start', self, response, 200, Page.START)
+
+    @task(2)
+    def enter_postcode(self):
+        """
+        POST postcode
+        """
+        with self.client.post("/en/requests/access-code/enter-address/", {
+            'form-enter-address-postcode': self.case['postcode']
+        }, catch_response=True) as response:
+            self.address_to_select = extractAddress(response, self.case["uprn"])
+            verify_response('RequestUacPost-EnterAddress', self, response, 200, Page.SELECT_ADDRESS,
+                            self.case["postcode"])
+
+    @task(3)
+    def select_address(self):
+        """
+        POST uprn and whole address extracted as JSON from the HTML in previous task
+        """
+        logger.info("Address selected: " + self.address_to_select)
+        with self.client.post("/en/requests/access-code/select-address/", {
+            'form-select-address': self.address_to_select
+        }, catch_response=True) as response:
+            verify_response('RequestUacPost-SelectAddress', self, response, 200, Page.ADDRESS_CORRECT, self.case["addressLine1"])
+
+    @task(4)
+    def confirm_address(self):
+        """
+        POST 'yes' to confirm address
+        """
+        with self.client.post("/en/requests/access-code/confirm-address/", {
+            'form-confirm-address': 'yes'
+        }, catch_response=True) as response:
+            verify_response('RequestUacPost-ConfirmAddress', self, response, 200, Page.SELECT_METHOD, "Post")
+
+    @task(5)
+    def select_method(self):
+        """
+        POST 'post' to select post as method of sending UACs
+        """
+        with self.client.post("/en/requests/access-code/select-method/", {
+            'form-select-method': 'post'
+        }, catch_response=True) as response:
+            verify_response('RequestUacPost-SelectMethod', self, response, 200, Page.ENTER_NAME, "")
+
+    @task(6)
+    def enter_name(self):
+        """
+        POST 'John' as first name and 'Smith' as last name of person to send the UAC to
+        """
+        with self.client.post("/en/requests/access-code/enter-name/", {
```

Comment:
As it currently stands the same name is being used for each simulated user. If RHUI had a threading bug then Locust wouldn't detect this, as the the name used by a different thread matches this one.
If not too expensive a randomly generated name would be good.

Reply:
Yes, I know this isn't ideal. But I think to make it do that would take me too long at the moment. I might come back to it after I've made changes to the SampleGenerator (unless it would be better to make the SampleGenerator create some names)?

Reply:
Getting the sampleGenerator is a good idea. It can generate a random first-name, surname & phone number.
Perhaps the memory impact is not too bad. We have been running with 3 million case/uacs and with 40 odd workers. That is 75,000 records per worker. So if it adds 25 chars for each record that'll be about 2mb.
Suggest 5 chars for a first name and 6 for a surname. Both with capitalised first character.
If you agree can you add extend the sampleGenerator CR to include this feature.

Reply:
Thanks Peter, I'm happy to add that to the CR. Do the names need to be readable or can they just be sequences of random letters?

Reply:
Thanks Peter, I'm happy to add that to the CR. Do the names need to be readable or can they just be sequences of random letters?

Reply:
now added to CR-1379

Reply:
I've also added a TODO to the code

Reply:
The name is now read in from the test data - I've removed the TODO